libimage/push.go

```diff
@@ -84,6 +86,12 @@ func (r *Runtime) Push(ctx context.Context, source, destination string, options
 		destRef = dockerRef
 	}
 
+	// docker-archive and only DockerV2Schema2MediaType support Gzip compression
+	if options.CompressionFormat != nil &&
+		(destRef.Transport().Name() == "docker-archive" || options.ManifestMIMEType == manifest.DockerV2Schema2MediaType) {
+		options.CompressionFormat = &compression.Gzip
```

Comment:
Just set it to `nil`.

This hard-codes the knowledge either way, but the `gzip` value is misleading in the case of `docker-archive`/`docker-daemon` (we output the layers uncompressed)